code/modules/mob/living/basic/lavaland/node_drone/node_drone.dm

```diff
@@ -0,0 +1,149 @@
+#define FLY_IN_STATE 1
+#define FLY_OUT_STATE 2
+#define NEUTRAL_STATE 3
+
+/**
+ * Mining drones that are spawned when starting a ore vent's wave defense minigame.
+ * They will latch onto the vent to defend it from lavaland mobs, and will flee if attacked by lavaland mobs.
+ * If the drone survives, they will fly away to safety as the vent spawns ores.
+ * If the drone dies, the wave defense will fail.
+ */
+
+/mob/living/basic/node_drone
+	name = "NODE drone"
+	desc = "Standard in-atmosphere drone, used by Nanotrasen to operate and excavate valuable ore vents."
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "mining_node_active"
+	icon_living = "mining_node_active"
+	icon_dead = "mining_node_active"
+
+	maxHealth = 500
+	health = 500
+	density = TRUE
+	pass_flags = PASSTABLE|PASSGRILLE|PASSMOB
+	mob_size = MOB_SIZE_LARGE
+	mob_biotypes = MOB_ROBOTIC
+	faction = list(FACTION_STATION, FACTION_NEUTRAL)
+	light_range = 4
+
+	speak_emote = list("chirps")
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "clangs"
+	response_harm_simple = "clang against"
+
+	ai_controller = /datum/ai_controller/basic_controller/node_drone
+
+	/// Is the drone currently attached to a vent?
+	var/active_node = FALSE
+	/// What status do we currently track for icon purposes?
+	var/flying_state = NEUTRAL_STATE
+	/// Weakref to the vent the drone is currently attached to.
+	var/obj/structure/ore_vent/attached_vent = null
+	/// Set when the drone is begining to leave lavaland after the vent is secured.
+	var/escaping = FALSE
+
+/mob/living/basic/node_drone/death(gibbed)
+	. = ..()
+	explosion(origin = src, light_impact_range = 1, smoke = 1)
+	qdel(src)
```

Comment:
probably just give it `basic_mob_flags = DEL_ON_DEATH`, might be some other things checking for that